pkg/segment/writer/segwriter.go

```diff
@@ -360,8 +360,9 @@ func updateValuesFromConfig() {
 func timeBasedWIPFlushToFile() {
 	for {
 		sleepDuration := time.Duration(config.GetSegFlushIntervalSecs()) * time.Second
+		maxFlushDuration := time.Duration(config.GetSegWipFlushMaxIntervalSecs()) * time.Second
 		time.Sleep(sleepDuration)
-		FlushWipBufferToFile(&sleepDuration)
+		FlushWipBufferToFile(&sleepDuration, &maxFlushDuration)
```

Comment:
do we want to overload the existing go routine to do both things ?
what happens if these two durations are different. We may miss out on the old one ?
we have two options:
1. create seperate go routines and each sleeps for it's own interval 
2. Keep one go routine and take the minimum of these two intervals and sleep for that duration
in both cases the `AppendWip..` call should still be called validating the timings

Reply:
Kunal, why would we miss out on the old one? I did not understand that.

Also I did not create a separate go routine, because that go routine also does the same exact functionality but instead of `lastUpdatedTime` it checks for `lastWipFlushTime`. Its like the min and max interval for a WIP flush block. So isn't better to do it in the same go routine?

Reply:
I have created two separate go routines, as we might miss out on `maxWaitWipFlushTime` if the intervals were close to each other.